challenges/cloud/sre.md

```diff
@@ -0,0 +1,376 @@
+# Summary
+
+Monitor, deploy, and extend a prototype service that provides an API for Kubernetes resources
+
+# Rationale
+
+This exercise has two goals:
+
+* It helps us to understand what to expect from you as a Site Reliability Engineer,
+  how you reason about a production service, how you reason about system design, your
+  opinions on automation & tooling, and how you communicate when trying to solve problems.
+* It helps you get a feel for what it would be like to work at Teleport, as this
+  exercise aims to simulate our day-as-usual and expose you to the type of work
+  we're doing here.
+
+We believe this technique is not only better, but also is more fun compared to
+whiteboard/quiz interviews, so common in the industry. It's not without the
+downsides - it could take longer than traditional interviews.
+
+[Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to hacking on this project together.
+
+# Requirements
+
+This challenge covers 5 engineering levels at Teleport. Level 6 is only for internal promotions. Check the
+[engineering levels document](https://raw.githubusercontent.com/gravitational/careers/main/levels.pdf)
+for more details.
+
+* Your submission will need to meet the requirements of the level you are applying for.
+* Split the submission into 2-3 pull requests for us to review. We will review
+  every pull request and provide our feedback.
+* We are going to inspect the repository, test it, and get back to you.
+
+## Server
+
+An incomplete implementation of a Go server is located [here](sre-server.go).
+You will extend this implementation by adding features described below depending
+on the level you are applying to. Start by creating a new Github repository. Then
+let the interview panel know the repository's location by pasting a link in your
+interview Slack channel. Invite interview panel participants as contributors to
+the new repository if you prefer to keep your submission private.
+
+## Design Doc
+
+Start with a brief design document that covers the edge cases and design approach. At
+Teleport, we prefer Markdown for
+[our designs](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md).
+
+Be sure to cover the following in your design:
```

Comment:
Reading this top to bottom, and by the time I get here I still have very little context as to what the thing I'm designing is supposed to do. The only information I've been given so far is the initial sentence that mentions API and Kubernetes.

Would it make sense to add a few more sentences about the project here so the reader has this context in mind when reading about what needs to go in the design doc?

Reply:
I see the issue that Zac is talking about. I'll add an intro section that summarizes the task.